src/ray/core_worker/context.h

```diff
@@ -130,6 +132,10 @@ class WorkerContext {
   std::shared_ptr<rpc::RuntimeEnvInfo> runtime_env_info_ GUARDED_BY(mutex_);
   /// The id of the (main) thread that constructed this worker context.
   const boost::thread::id main_thread_id_;
+  /// The currently executing main thread's task id. It's the actor creation task id
+  /// for actor, or the main thread's task id for normal tasks.
+  /// Used merely for observability purposes to track task hierarchy.
+  TaskID main_thread_or_actor_creation_task_id;
```

Comment:
```suggestion
  TaskID main_thread_or_actor_creation_task_id_;
```

And we may need to guard it with mutex